PWGCF/JCorran/Tasks/jEPDzeroFlowAnalysis.cxx

```diff
@@ -0,0 +1,212 @@
+  ConfigurableAxis centAxis{"centAxis", {VARIABLE_WIDTH, 0, 10, 20, 30, 40, 50, 60, 70, 80, 100}, "Centrality interval"};
+  ConfigurableAxis cosAxis{"cosAxis", {110, -1.05, 1.05}, "Cosine axis"};
+
+  //  Filter track2pFilter = (nabs(aod::cf2prongtrack::eta) < cfgEtaMax) && (aod::cf2prongtrack::pt > cfgPtMin);
+
+  EventPlaneHelper helperEP;
+
+  int DetId;
+  int RefAId;
+  int RefBId;
+
+  float centrality;
+
+  template <class T>
+  using hasInvMass = decltype(std::declval<T&>().invMass());
+
+  template <typename T>
+  int GetDetId(const T& name)
+  {
+    if (name.value == "FT0C") {
+      return 0;
+    } else if (name.value == "FT0A") {
+      return 1;
+    } else if (name.value == "FT0M") {
+      return 2;
+    } else if (name.value == "FV0A") {
+      return 3;
+    } else if (name.value == "TPCPos") {
+      return 4;
+    } else if (name.value == "TPCNeg") {
+      return 5;
+    } else if (name.value == "TPCTot") {
+      return 6;
+    } else {
+      return 0;
+    }
+  }
+
+  template <typename TCollision>
+  bool eventSelected(TCollision collision)
+  {
+    if (!collision.sel8()) {
+      return 0;
+    }
+    if (cfgCentSel < centrality) {
+      return 0;
+    }
+    if (!collision.selection_bit(aod::evsel::kIsGoodZvtxFT0vsPV)) {
+      return 0;
+    }
+    if (!collision.selection_bit(aod::evsel::kNoSameBunchPileup)) {
+      return 0;
+    }
+    if (cfgPVSel && std::abs(collision.posZ()) > cfgPV) {
+      return 0;
+    }
+    if (cfgAddEvtSelPileup && !collision.selection_bit(o2::aod::evsel::kNoCollInTimeRangeStandard)) {
+      return 0;
+    }
+    if (collision.trackOccupancyInTimeRange() > cfgMaxOccupancy || collision.trackOccupancyInTimeRange() < cfgMinOccupancy) {
+      return 0;
+    }
+    return 1;
+  } // event selection
+
+  template <typename CollType, typename TrackType>
+  void fillHistosFlow(const CollType& coll, TrackType& trks)
+  {
+    if (coll.qvecAmp()[DetId] < 1e-4 || coll.qvecAmp()[RefAId] < 1e-4 || coll.qvecAmp()[RefBId] < 1e-4) {
+      return;
+    }
+    int DetInd = DetId * 4 + cfgNQvec * 4;
+    //    int RefAInd = RefAId * 4 + cfgNQvec * 4;
+    //    int RefBInd = RefBId * 4 + cfgNQvec * 4;
+    for (auto& trk : trks) {
+      histos.fill(HIST("hist_EP_cos_Det_v2"), trk.invMass(), trk.pt(), std::cos(2.0 * (trk.phi() - helperEP.GetEventPlane(coll.qvecRe()[DetInd + 3], coll.qvecIm()[DetInd + 3], 2))), centrality);
+      histos.fill(HIST("hist_EP_sin_Det_v2"), trk.invMass(), trk.pt(), std::sin(2.0 * (trk.phi() - helperEP.GetEventPlane(coll.qvecRe()[DetInd + 3], coll.qvecIm()[DetInd + 3], 2))), centrality);
+    }
+  }
+
+  void init(InitContext const&)
+  {
+    DetId = GetDetId(cfgDetName);
+    RefAId = GetDetId(cfgRefAName);
+    RefBId = GetDetId(cfgRefBName);
+
+    if (DetId == RefAId || DetId == RefBId || RefAId == RefBId) {
+      LOGF(info, "Wrong detector configuration \n The FT0C will be used to get Q-Vector \n The TPCpos and TPCneg will be used as reference systems");
```

Comment:
Shouldn't be better to have a `fatal` here to enforce the user to change the configuration?